### PR TITLE
Cleanup readme, license and build files.

### PR DESCRIPTION
**Pull request by bogdandrutu**
A lot of small fixes:
* Update license to use OpenCensus Authors;
* Update project name to be OpenCensus;
* Update project groupId to be io.opencensus (we may have to change it to io.opencensus.proto later);
* Rename BUILD -> BUILD.bazel to fix TODO from gradle.
* Update protoc to 3.4.0

### CODE REVIEW

File: BUILD.bazel [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package(default_visibility = ["//visibility:public"])
```
Comment (songy23) [inline BUILD.bazel]: Does this file need a copyright header?

Comment (bogdandrutu) [reply BUILD.bazel]: grpc-java does not have, grpc (c++) has. I don't know, any suggestions?

Comment (songy23) [reply BUILD.bazel]: I'd suggest to keep the header as before.

Comment (bogdandrutu) [reply BUILD.bazel]: Done.

**Pull request merged**